src/components/application_manager/src/hmi_state.cc

```diff
@@ -241,38 +265,123 @@ AudioSource::AudioSource(std::shared_ptr<Application> app,
+    return std::max(parent()->hmi_level(), max_hmi_level());
+  }
+
+  auto expected = HMILevel::HMI_BACKGROUND;
+  if (is_navi_app()) {
+    expected = HMILevel::HMI_LIMITED;
+  }
+
+  return std::max(std::max(expected, max_hmi_level()), parent()->hmi_level());
+}
+
+mobile_apis::AudioStreamingState::eType AudioSource::audio_streaming_state()
+    const {
+  return is_media_app() ? mobile_apis::AudioStreamingState::NOT_AUDIBLE
+                        : parent()->audio_streaming_state();
+}
+
+mobile_apis::VideoStreamingState::eType AudioSource::video_streaming_state()
+    const {
+  return parent()->video_streaming_state();
+}
+
+mobile_apis::HMILevel::eType AudioSource::max_hmi_level() const {
+  using namespace mobile_apis;
+
+  auto parent_max = parent() ? parent()->max_hmi_level() : HMILevel::HMI_FULL;
   if (WindowType::WIDGET == window_type()) {
-    return parent()->hmi_level();
+    return std::max(HMILevel::HMI_FULL, parent_max);
   }
 
-  // Checking for NONE  is necessary to avoid issue during
-  // calculation of HMI level during setting default HMI level
-  if (keep_context_ || HMILevel::HMI_NONE == parent()->hmi_level()) {
-    return parent()->hmi_level();
+  auto expected = HMILevel::HMI_FULL;
+  if (!keep_context_ && is_media_app()) {
+    expected = HMILevel::HMI_BACKGROUND;
   }
 
-  return HMILevel::HMI_BACKGROUND;
+  return std::max(expected, parent_max);
+}
+
+mobile_apis::AudioStreamingState::eType AudioSource::max_audio_streaming_state()
+    const {
+  auto parent_max = parent() ? parent()->max_audio_streaming_state()
+                             : mobile_apis::AudioStreamingState::AUDIBLE;
+  return is_media_app() ? mobile_apis::AudioStreamingState::NOT_AUDIBLE
+                        : parent_max;
+}
+
+mobile_apis::VideoStreamingState::eType AudioSource::max_video_streaming_state()
+    const {
+  auto parent_max = parent() ? parent()->max_video_streaming_state()
+                             : mobile_apis::VideoStreamingState::STREAMABLE;
+  return parent_max;
 }
 
 EmbeddedNavi::EmbeddedNavi(std::shared_ptr<Application> app,
                            const ApplicationManager& app_mngr)
     : HmiState(app, app_mngr, STATE_ID_EMBEDDED_NAVI) {}
 
+mobile_apis::AudioStreamingState::eType EmbeddedNavi::audio_streaming_state()
+    const {
+  return is_navi_app() ? mobile_apis::AudioStreamingState::NOT_AUDIBLE
+                       : parent()->audio_streaming_state();
+}
+
+mobile_apis::VideoStreamingState::eType EmbeddedNavi::video_streaming_state()
+    const {
+  return is_navi_app() ? mobile_apis::VideoStreamingState::NOT_STREAMABLE
+                       : parent()->video_streaming_state();
+}
+
 mobile_apis::HMILevel::eType EmbeddedNavi::hmi_level() const {
   using namespace mobile_apis;
   using namespace helpers;
 
   if (WindowType::WIDGET == window_type()) {
-    return parent()->hmi_level();
+    return std::max(parent()->hmi_level(), max_hmi_level());
```

Comment:
do we need to check parent() is valid here? we do in `max_video_streaming_state`